docs/workspaces/index.md

```diff
@@ -156,11 +156,27 @@ A minimal `question.html` for an externally graded workspace should look somethi
 </pl-submission-panel>
 ```
 
-### Creating files in the workspace home directory
+## Creating files in the workspace home directory
 
 Workspace questions can optionally include a `workspace/` subdirectory within the regular [PrairieLearn question directory structure](../question.md#directory-structure). If this `workspace/` subdirectory exists, its contents will be copied into the home directory of the student's workspace container, as configured in the `home` setting in `info.json`.
 
-Questions using workspaces can also be randomized, i.e., include files that contain random and dynamic content. This is done using [the `server.py` file in the question directory](../question.md#question-serverpy). In addition to other random parameters that can be created for the question page itself, the `_workspace_files` parameter can also be set, containing an array of files to be created. Each element of the array must include a `name` property, containing the file name (which can include a path with directories), and either a `contents` property, containing the contents of the file, or a `questionFile` property, pointing to an existing file in a different location in the question directory. For example:
+Questions using workspaces can also be randomized, i.e., include files that contain random and dynamic content. This may be done in two ways: using mustache-based template files or using [the `server.py` file in the question directory](../question.md#question-serverpy). For template files, a workspace question can optionally include a `workspaceTemplates/` subdirectory within the regular question directory structure. The contents will be copied into the home directory of the student's workspace container, as with the `workspace/` directory. However, all files within this directory may include mustache tags (e.g., `{{params.value}}`), which will be replaced with the equivalent values set by `server.py`. For example, if `server.py` sets `data["params"]["starting_value"]` to `17`, then if the file `main.py` inside `workspaceTemplates` has the following content:
```

Comment:
Recommendation: support an optional `.mustache` extension on files, and strip off that suffix if it exists. This will help folks like me who get endlessly annoyed when my editor complains about invalid syntax in templated files. In other words, the below file could be `main.py.mustache`, and it would appear to the workspace as `main.py`.

If someone really wants a templated mustache file that is itself a mustache file (that hurts my brain), they could us `main.mustache.mustache`.

Reply:
Done in 9b3a54b9b